### PR TITLE
Major refactor of code to use HazenTask classes

### PR DESCRIPTION
**Pull request by laurencejackson**
Following work to separate the webapp and hazenlib functionality of the hazen project the decision was made to take this opportunity to standardise hazen tasks.

This led to the introduction of the HazenTask class, all tasks that hazen performs will inherit from this superclass leading to a number of benefits:

- allows us to easily add new tasks without rewriting boilerplate on new tasks
- the webapp is able to automatically find relevant tasks by searching for HazenTask objects
- reduces code duplication
- standardised data input/output across all tasks using HazenTask
- separating the webapp and hazenlib features massively simplifies the repository install process for most users

Currently, all MagNet object tasks currently performed as part of routine QA at GSTT have been refactored to use the HazenTask syntax. The following tasks have not been converted to the HazenTask syntax but are still available on the CLI interface:

snr_map @pcw24601 
relaxometry @pcw24601 
acr_uniformity @YassineRMH 

These tasks should be converted to the new syntax when possible to allow them to be accessible on the web frontend.

### CODE REVIEW

File: .github/workflows/tests_development.yml [hunk 1]
```diff
@@ -18,25 +18,27 @@ jobs:
...
+#          POSTGRES_DB: hazen_test
+#          POSTGRES_USER: test_user
+#          POSTGRES_PASSWORD: test_user_password
+#        # Set health checks to wait until postgres has started
+#        options: >-
+#          --health-cmd pg_isready
+#          --health-interval 10s
+#          --health-timeout 5s
+#          --health-retries 5
+
```
Comment (tomaroberts) [inline .github/workflows/tests_development.yml]: This is commented out in `tests_development.yml` but not `tests_release.yml` – is that intentional or an oversight?

Comment (laurencejackson) [reply .github/workflows/tests_development.yml]: oversight, will correct now

File: newdocs/source/contributors.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+* `Lucrezia Cester <https://github.com/Lucrezia-Cester>`_
+* `Paul Wilson <https://github.com/pcw24601>`_
+* `Yassine Azma <https://github.com/YassineRMH>`_
+* `Haleema Al Jazzaf <https://github.com/heyhaleema>`_
+* `Sophie Ratkai <https://github.com/sophie22>`_
+* `Ryan Satnarine <https://github.com/rs-sprout98>`_
+* `Jane Ansell <https://github.com/ansellj>`_
+* `Jamie Small <https://github.com/JamieSmall>`_
+* `Elizabeth Stamou <https://github.com/elizaGSTT>`_
+* `Elizabeth Stamou <https://github.com/elizaGSTT>`_
```
Comment (tomaroberts) [inline newdocs/source/contributors.rst]: So good we included her twice

**Comment by tomaroberts** [comment]
Tested:
- CLI version installed via PyPI – worked
- CLI version installed via `pip install -r requirements.txt` - worked and unit tests successful (albeit with some usual warnings)
- new `--output` option – worked

Broadly reviewed other files.

Good to go.

**Pull request merged**